sdk/core/azure-core-amqp/src/impl/rust_amqp/amqp/claim_based_security.cpp

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "private/claims_based_security_impl.hpp"
+#include "private/connection_impl.hpp"
+#include "private/management_impl.hpp"
+#include "private/session_impl.hpp"
+
+#include <azure/core/diagnostics/logger.hpp>
+#include <azure/core/internal/diagnostics/log.hpp>
+
+using namespace Azure::Core::Diagnostics::_internal;
+using namespace Azure::Core::Diagnostics;
+
+namespace Azure { namespace Core { namespace Amqp { namespace _detail {
+  using namespace Azure::Core::Amqp::_internal;
+
+  ClaimsBasedSecurityImpl::ClaimsBasedSecurityImpl(std::shared_ptr<_detail::SessionImpl> session)
```

Comment:
For the review, I am going to assume most of these file moves/refactors leave the code unchanged. If there's anything interesting that has changed and worth calling attention to it, please share.